### DISCUSSION TITLE
After 3 years of use, there are now 6K transactions, the problem is that the transactions are particularly slow to open, is it possible to optimize.

### DISCUSSION THREAD

**Discussion by wqwwffsytxwp** [firefly-iii/firefly-iii#9017] [General]
Debug information generated at 2024-07-03 13:43:18 for Firefly III version **v6.1.18**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.18 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.8 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td>mysql, pgsql, <strong>*sqlite*</strong>, </td></tr><tr><td>Docker build</td><td><span>#</span>1104, base <span>#</span>85</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Asia/Shanghai + Asia/Shanghai</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>info, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>zh_CN + equal</td></tr><tr><td>Trusted proxies</td><td>*</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-07-02 19:00:07 (10 hours ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has rules">:wrench:</span> <span title="Has recurring transactions">:clock130:</span></td></tr><tr><td>Session start</td><td>2024-07-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-07-03 23:59:59</td></tr><tr><td>View range</td><td>MTD</td></tr><tr><td>User language</td><td>zh_CN</td></tr><tr><td>User locale</td><td>zh_CN</td></tr><tr><td>Locale(s) supported</td><td>zh_CN.utf8: :white_check_mark:<br>zh_CN.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0</td></tr></tbody></table>
After 3 years of use, there are now 6K transactions, the problem is that the transactions are particularly slow to open, is it possible to optimize.

**Comment by JC5** [comment]
Some pages may be slow, but it also depends on your system. I am working on optimizations.

**Reply by gamebeaker** [reply]
One database optimization could be timescaledb [https://www.timescale.com/](https://www.timescale.com/) (shard intervall 1 month or even one year) it is postgresql with an extension. The SQL commands don't change. It optimizes in the background. Here is a [tutorial](https://www.youtube.com/watch?v=vm_hcP4CDbA&list=PLsceB9ac9MHScvW5NBuCaYafW87hP-Gi2&index=1) playlist where they use docker.
But i find it weird that 6k transations should slow a database down. Is there a known reason why that is the case?
@wqwwffsytxwp are you trying to display all transactions or only the current month?
